h/api/models/token.py

```diff
@@ -0,0 +1,38 @@
+# -*- coding: utf-8 -*-
+import uuid
+
+import sqlalchemy
+
+from h.api import db
+
+
+class Token(db.Base):
+
+    """A long-lived API token for a user."""
+
+    __tablename__ = 'token'
+    userid = sqlalchemy.Column(sqlalchemy.UnicodeText(),
+                               primary_key=True)
+    token = sqlalchemy.Column(sqlalchemy.UnicodeText(),
+                              unique=True,
+                              nullable=False)
+
+    def __init__(self, userid):
+        self.userid = userid
+        self.token = 'dev-' + str(uuid.uuid4())
+
+    @classmethod
+    def get(cls, userid):
+        return cls.query.get(userid)
+
+    @classmethod
+    def userid_for_token(cls, token):
```

Comment:
As mentioned earlier, class methods on models are usually getters/searchers which return model objects. Having one return a property is a bit odd.